scipy/optimize/_differentialevolution.py

```diff
@@ -598,6 +625,41 @@ def __init__(self, func, bounds, args=(),
 
         self.random_number_generator = check_random_state(seed)
 
+        # Which parameters are going to be integers?
+        if np.any(integrality):
+            # # user has provided a truth value for integer constraints
+            # if polish:
+            #     message = ("Polishing is incompatible with integrality"
+            #                " constraints; `polish` will be ignored.")
+            #     self.polish = False
+            #     warnings.warn(message, OptimizeWarning)
+
```

Comment:
```suggestion
```
Yeah, this is fine. I had planned on doing what you've done here in `brute`, too, I just left it out of the initial commit because I didn't want to test it : P Really just trying to push what I _needed_ for fitting discrete distributions. (Polishing isn't as important there because the iterative/zooming `brute` thing does a pretty good job of polishing the solution itself.)